### ISSUE TITLE
Replace Grunt with Gulp

### ISSUE THREAD

**Issue by jsalahov** [Rostlab/JS16_ProjectD_Group4#22]
Guys, if you do not have any sincere feelings towards Grunt :stuck_out_tongue_winking_eye: can I replace it with Gulp? :smiling_imp:

**Comment by julienschmidt** [comment]
What's the purpose? Grunt runs great so far.

**Comment by jsalahov** [comment]
Of course it does. And I can assure you, so will Gulp :smile: I just need to add some stuff related to tests, don't want to mess with grunt

**Comment by julienschmidt** [comment]
If you configure jshint with the same settings again (and adjust the Travis cfg), I'm fine with it.

**Comment by jsalahov** [comment]
Yeah, sure

**Comment by jsalahov** [comment]
@julienschmidt As I see it, precommit hook is set up as optional step. Is this intentional? I can make it mandatory/automatic, if needed.

**Comment by julienschmidt** [comment]
It's an optional client-side hook. You could probably enforce it with gulp but that's a pretty bad idea in my opinion, you should never mess with a client's git config without asking first.

**Comment by julienschmidt** [comment]
Also it seems to break GitHub desktop since the env is not configured correctly there.

**Comment by jsalahov** [comment]
Just asking, will leave it as optional step

**Comment by sacdallago** [comment]
I like the idea of the hook btw

**Comment by sacdallago** [comment]
P.S.: didn't mean to close the issue. :baby:

**Comment by jsalahov** [comment]
@julienschmidt It turns out Github desktop doesn't support hooks. So you have to commit from bash for hooks to work

**Comment by julienschmidt** [comment]
It doesn't have to, since it just runs git and git supports those hooks :stuck_out_tongue_closed_eyes: 
But npm needs some environment vars which are not available when it's called from Github Desktop.

I use git from the shell anyway :smile:  (But I don't expect anyone else to)

**Comment by jsalahov** [comment]
Well, it  doesn't run git (at least on Windows), crazy I know :smile: . It uses a library called libgit

**Comment by julienschmidt** [comment]
Actually I just got it working with GitHub Desktop (Mac) with as slightly extended hook script:

``` sh
#!/usr/bin/env sh
PATH=$PATH:/usr/local/bin:/usr/local/sbin
npm test
```

**Comment by jsalahov** [comment]
I just read about that :smile: Well, I'm never gonna get those shiny ui things(and their problems :wink: )  anyway. Linux FTW

**Comment by julienschmidt** [comment]
Fixed by #27
